src/managers/cluster.py

```diff
@@ -26,3 +37,93 @@ def get_host_mapping(self) -> dict[str, str]:
         ip = socket.gethostbyname(hostname)
 
         return {"hostname": hostname, "ip": ip}
+
+    def get_leader(self) -> str:
+        """Query the etcd cluster for the raft leader and return the client_url as string."""
+        leader = ""
+
+        # loop through list of hosts and compare their member id with the leader
+        # if they match, return this host's endpoint
+        for endpoint in self.cluster_endpoints:
+            client = EtcdClient(client_url=endpoint)
+            try:
+                endpoint_status = client.get_endpoint_status()
+                member_id = endpoint_status["Status"]["header"]["member_id"]
+                leader_id = endpoint_status["Status"]["leader"]
+                if member_id == leader_id:
+                    leader = endpoint
+                    break
+            except KeyError:
```

Comment:
Is there a case where no leader is found?

Reply:
Yes, this could happen during leader election or in error cases.